definition/IRoom.ts

```diff
@@ -76,13 +76,26 @@ export interface ICreatedRoom extends IRoom {
 	rid: string;
 }
 
+export interface ITeamRoom extends IRoom {
+	teamMain: boolean;
+	teamId: string;
+}
+
+export const isTeamRoom = (room: Partial<IRoom>): room is ITeamRoom => !!room.teamMain;
+export const isPrivateTeamRoom = (room: Partial<IRoom>): room is ITeamRoom => isTeamRoom(room) && room.t === 'p';
+export const isPublicTeamRoom = (room: Partial<IRoom>): room is ITeamRoom => isTeamRoom(room) && room.t === 'c';
+
+export const isDiscussion = (room: Partial<IRoom>): room is ITeamRoom => !!room.prid;
+export const isPrivateDiscussion = (room: Partial<IRoom>): room is ITeamRoom => isDiscussion(room) && room.t === 'p';
+export const isPublicDiscussion = (room: Partial<IRoom>): room is ITeamRoom => isDiscussion(room) && room.t === 'c';
```

Comment:
why are those functions type guarding to ITeamRoom?

Reply:
Good catch, thanks!